File: Examples/DeveloperObjcExample/ObjcExample/ProfileViewController.m
//
//  ProfileViewController.m
//  AppcuesObjcExample
//
//  Created by Matt on 2022-06-01.
//

#import "ProfileViewController.h"
#import "Appcues+Shared.h"
#import "SignInViewController.h"

@interface ProfileViewController ()
@property (weak, nonatomic) IBOutlet UITextField *givenNameTextField;
@property (weak, nonatomic) IBOutlet UITextField *familyNameTextField;

@end

@implementation ProfileViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];

    [[Appcues shared] screenWithTitle:@"Update Profile" properties:nil];
}

- (IBAction)saveButtonTapped:(UIButton *)sender {
    [self.view endEditing:YES];

    NSMutableDictionary<NSString *, NSString *> *properties = [NSMutableDictionary dictionary];

    if (_givenNameTextField.text != nil && _givenNameTextField.text.length != 0) {
        properties[@"givenName"] = _givenNameTextField.text;
    }

    if (_familyNameTextField.text != nil && _familyNameTextField.text.length != 0) {
        properties[@"familyName"] = _familyNameTextField.text;
    }


    [[Appcues shared] identifyWithUserID:currentUserID properties:properties];

    _givenNameTextField.text = nil;
    _familyNameTextField.text = nil;
}

@end

File: Examples/DeveloperCocoapodsExample/CocoapodsExample/ProfileViewController.swift
//
//  ProfileViewController.swift
//  AppcuesExample
//
//  Created by Matt on 2021-10-12.
//

import UIKit
import AppcuesKit

class ProfileViewController: UIViewController {

    @IBOutlet private var givenNameTextField: UITextField!
    @IBOutlet private var familyNameTextField: UITextField!

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }

    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)

        Appcues.shared.screen(title: "Update Profile")
    }

    @IBAction private func saveButtonTapped(_ sender: UIButton) {
        view.endEditing(true)

        var properties: [String: String] = [:]

        if let givenName = givenNameTextField.text, !givenName.isEmpty {
            properties["givenName"] = givenName
        }

        if let familyName = familyNameTextField.text, !familyName.isEmpty {
            properties["familyName"] = familyName
        }

        Appcues.shared.identify(userID: User.currentID, properties: properties)

        givenNameTextField.text = nil
        familyNameTextField.text = nil
    }
}
